### ISSUE TITLE
Not working properly with jsdoc and @typescript-eslint plugins

### ISSUE THREAD

**Issue by johnn-guerrero-aurea** [paleite/eslint-plugin-diff#13]
Love this library, very useful for linting only modified files.

We are having an issue where modified files are detected, but when this plugin is applied there are some rules skipped.

Configuration without `diff` plugin
```
{
  "root": true,
  "plugins": [
    "jsdoc",
    "@typescript-eslint"
  ],
  "parser": "@typescript-eslint/parser",
  "extends": [
    "plugin:jsdoc/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/eslint-recommended"
  ],
  "rules": {
    "jsdoc/require-jsdoc": "error"
  }
}
```

importTopics.ts file
![image](https://user-images.githubusercontent.com/36961706/128368613-5843fad1-eec1-4e4a-af9f-cd597de2e7a1.png)

Lint Result:
![image](https://user-images.githubusercontent.com/36961706/128368877-a29d5657-70c0-4067-aa7c-6178579712d5.png)

---
Configuration with `diff` plugin
```
{
  "root": true,
  "plugins": [
    "jsdoc",
    "@typescript-eslint"
  ],
  "parser": "@typescript-eslint/parser",
  "extends": [
    "plugin:jsdoc/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:diff/diff"
  ],
  "rules": {
    "jsdoc/require-jsdoc": "error"
  }
}
```

importTopics.ts file
![image](https://user-images.githubusercontent.com/36961706/128368613-5843fad1-eec1-4e4a-af9f-cd597de2e7a1.png)

Lint Result:
![image](https://user-images.githubusercontent.com/36961706/128367440-70f7358f-ba1c-4ab0-907b-05e013e89b01.png)

Notice that after the plugin is applied we get only one rule violation, others are simply skipped. These are important rules in our process, so to continue using your awesome library we need to have this fixed.

We want ALL rules defined in the 'extends' section to be applied to modified files.

Is there any configuration I could be missing?
Or is this an issue with the library working along with these other plugins?

Thanks in advance for your help!

**Comment by paleite** [comment]
Hi @johnn-guerrero-aurea ,
thanks for your thorough report!

I think this may be a misunderstanding, but please let me know if I didn't understand properly. You mentioned that you use the plugin to lint modified _files_, but this plugin is designed to be even more granular; its purpose is to only report modified _lines_, so any errors outside of the green lines in your screenshot will be omitted by this plugin.

This is useful when you only want to check the code that you've written. E.g. when you lint in a pre-commit hook, only _your own_ errors will prevent the linter from passing.

Another use-case is for when you're introducing new rules, so that code that was considered passing before, will not suddenly make the build break.

**Comment by johnn-guerrero-aurea** [comment]
@paleite Thanks for your answer.

In the screenshots above, green lines are _new_ code. blue lines are _modified_ code. I would expect the plugin to work with both.

Is there any configuration missing? Thanks again for your time.

**Comment by paleite** [comment]
Oh, I think I get it now, but please correct me if I'm wrong:
- Without eslint-plugin-diff you're getting an error on line 48
- line 48 is modified
- With eslint-plugin-diff, the error on line 48 is dropped

If so, I agree, that's definitely a bug 🐛 
I was unable to reproduce it so far, but if you could provide a reproduction repo, I'd love to take a look at it!

I'll re-open this ticket and mark it as a bug.

Another thing: I notice that the images of importTopics.ts are identical, but I assume the line-results are correct.

**Comment by paleite** [comment]
And thanks again, @johnn-guerrero-aurea for taking the time to report this. I really appreciate it!

**Comment by paleite** [comment]
Closing this for inactivity: Can't reproduce
